src/my_crypto.erl

```diff
@@ -1,26 +1,112 @@
 -module(my_crypto).
 
+-include_lib("eunit/include/eunit.hrl").
+
 -export([hash/1, sign/2, verify/3, read_file_key/2]).
 
--spec hash(iodata()) -> binary().
-hash(Bin) -> crypto:hash(sha256, Bin).
+%% -define(MY_CRYPTO_DEBUG, true).
+
+%% Actual function implementations
 
--spec sign(binary(), public_key:private_key()) -> binary().
-sign(_Hash, _PrivateKey) -> <<>>. %%public_key:sign(Hash, none, PrivateKey).
+hash_full(Bin) -> crypto:hash(sha256, Bin).
 
--spec verify(binary(), binary(), public_key:public_key()) -> boolean().
-verify(_Hash, _Signature, _PubKey) -> true. %%public_key:verify(Hash, none, Signature, PubKey).
+sign_full(Hash, PrivateKey) -> public_key:sign(Hash, none, PrivateKey).
 
--spec read_file_key(private, string()) -> _.
-read_file_key(private, FileName) ->
+verify_full(Hash, Signature, PubKey) -> public_key:verify(Hash, none, Signature, PubKey).
+
+read_file_key_full(private, FileName) ->
     {ok, PemBin} = file:read_file(FileName),
     [RSAEntry] = public_key:pem_decode(PemBin),
-    _Key = public_key:pem_entry_decode(RSAEntry),
-    %% Key;
-    FileName;
+    Key = public_key:pem_entry_decode(RSAEntry),
+    Key;
 
-read_file_key(public, FileName) ->
+    
+read_file_key_full(public, FileName) ->
     {ok, Bin} = file:read_file(FileName),
-    [{_Key, _}] = public_key:ssh_decode(Bin, public_key),
-    %% Key;
-    FileName.
+    [{Key, _}] = public_key:ssh_decode(Bin, public_key),
+    Key.
+
+%% Debugging simplified functions
+
+%% cut hash down to three bytes
+hash_debug(Bin) -> 
+    Full = crypto:hash(sha256, Bin),
+    Part = binary:list_to_bin(binary:bin_to_list(Full, 0, 3)),
+
+    %% Copied this online :cat_scream:
+    HexString = lists:flatten([io_lib:format("~2.16.0B",[X]) || <<X:8>> <= Part ]),
+
+    HexString.
+
+drop_ext(FileName) ->
+    [Out | _] = string:split(FileName, "."),
+    Out.
+
+sign_debug(Hash, AnyKey) -> Hash ++ " " ++ drop_ext(AnyKey).
+verify_debug(Hash, Signature, PublicKey) -> Signature == sign_debug(Hash, PublicKey).
+read_file_key_debug(private, FileName) -> FileName;
+read_file_key_debug(public, FileName) -> FileName.
+
+-ifdef(MY_CRYPTO_DEBUG).
```

Comment:
😱

Reply:
gonna change it to POTATO_CRYPTOGRAPHY_MODE